src/main/java/org/jabref/gui/StateManager.java

```diff
@@ -201,4 +206,30 @@ public LastAutomaticFieldEditorEdit getLastAutomaticFieldEditorEdit() {
     public void setLastAutomaticFieldEditorEdit(LastAutomaticFieldEditorEdit automaticFieldEditorEdit) {
         lastAutomaticFieldEditorEditProperty().set(automaticFieldEditorEdit);
     }
+
+    public void addSearchHistory(String search) {
+        searchHistory.add(search);
+    }
+
+    public ObservableSet<String> getWholeSearchHistory() {
+        return searchHistory;
+    }
+
+    public List<String> getLastSearchHistory(int size) {
+        List<String> lastSearches = new ArrayList<>();
```

Comment:
Could you check if there can some code be shared as a new list type with the FileHistory class? I remember that I worked some time ago with a similar mechanic to limit the max number of entries in the list.